awx_collection/plugins/modules/tower_job_wait.py

```diff
@@ -90,78 +92,105 @@
-        else:
-            outfile = StringIO()
-        params['outfile'] = outfile
-
-        job_id = params.get('job_id')
-        try:
-            result = job.monitor(job_id, **params)
-        except exc.Timeout:
-            result = job.status(job_id)
-            result['id'] = job_id
-            json_output['msg'] = 'Timeout waiting for job to finish.'
-            json_output['timeout'] = True
-        except exc.NotFound as excinfo:
-            fail_json = dict(msg='Unable to wait, no job_id {0} found: {1}'.format(job_id, excinfo), changed=False)
-        except exc.JobFailure as excinfo:
-            fail_json = dict(msg='Job with id={0} failed, error: {1}'.format(job_id, excinfo))
-            fail_json['success'] = False
-            result = job.get(job_id)
-            for k in ('id', 'status', 'elapsed', 'started', 'finished'):
-                fail_json[k] = result.get(k)
-        except (exc.ConnectionError, exc.BadRequest, exc.AuthError) as excinfo:
-            fail_json = dict(msg='Unable to wait for job: {0}'.format(excinfo), changed=False)
-
-    if fail_json is not None:
-        module.fail_json(**fail_json)
-
-    json_output['success'] = True
-    for k in ('id', 'status', 'elapsed', 'started', 'finished'):
-        json_output[k] = result.get(k)
-
-    module.exit_json(**json_output)
+    # Create a module for ourselves
+    module = TowerModule(argument_spec=argument_spec, supports_check_mode=True)
+
+    # Extract our parameters
+    job_id = module.params.get('job_id')
+    timeout = module.params.get('timeout')
+    min_interval = module.params.get('min_interval')
+    max_interval = module.params.get('max_interval')
+
+    # Attempt to look up job based on the provided id
+    job = module.get_one('jobs', **{
+        'data': {
+            'id': job_id,
+        }
+    })
+
+    if job is None:
+        module.fail_json(msg='Unable to wait, on job {0} that ID does not exist in Tower.'.format(job_id))
+
+    job_url = job['url']
+
+    # This comes from tower_cli/models/base.py from the old tower-cli
+    dots = itertools.cycle([0, 1, 2, 3])
+    interval = min_interval
+    start = time.time()
+
+    # Poll the Ansible Tower instance for status, and print the status to the outfile (usually standard out).
+    result = check_job(module, job_url)
+
+    last_poll = time.time()
+    timeout_check = 0
+    while not result['finished']:
+        # Sanity check: Have we officially timed out?
+        # The timeout check is incremented below, so this is checking to see if we were timed out as of
+        # the previous iteration. If we are timed out, abort.
+        if timeout and timeout_check - start > timeout:
+            module.json_output['msg'] = "Monitoring aborted due to timeout"
+            module.fail_json(**module.json_output)
+
+        # If the outfile is a TTY, print the current status.
+        output = '\rCurrent status: %s%s' % (result['status'], '.' * next(dots))
```

Comment:
This var isn't used anywhere.

And we have no need to print dots. I think we can lose everything related to this

Reply:
Good catch, resolved.